src/main/java/com/iexec/sms/tee/session/base/SecretSessionBaseService.java

```diff
@@ -141,81 +125,98 @@ public Map<String, Object> getSecretsTokens(TeeSecretsSessionRequest request) th
+                .filter(e ->
+                // extract trusted en vars to include
+                trustedEnv.contains(e.getKey())
+                        // extract <IEXEC_INPUT_FILE_URL_N, url>
+                        || e.getKey().contains(IexecEnvUtils.IEXEC_INPUT_FILE_URL_PREFIX))
                 .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
-        tokens.put(INPUT_FILE_URLS, inputFileUrls);
-        return tokens;
+        tokens.putAll(trustedEnvVars);
+        return enclaveBase
+                .environment(tokens)
+                .build();
     }
 
     /*
      * Compute (App)
      */
-    public Map<String, Object> getAppTokens(TeeSecretsSessionRequest request)
-            throws TeeSessionGenerationException{
+    public SecretEnclaveBase getAppTokens(TeeSessionRequest request)
+            throws TeeSessionGenerationException {
+        SecretEnclaveBaseBuilder enclaveBase = SecretEnclaveBase.builder();
+        enclaveBase.name("app");
         TaskDescription taskDescription = request.getTaskDescription();
         if (taskDescription == null) {
             throw new TeeSessionGenerationException(
                     NO_TASK_DESCRIPTION,
-                    "Task description must no be null"
-            );
+                    "Task description must no be null");
         }
 
         Map<String, Object> tokens = new HashMap<>();
         TeeEnclaveConfiguration enclaveConfig = taskDescription.getAppEnclaveConfiguration();
         if (enclaveConfig == null) {
             throw new TeeSessionGenerationException(
                     APP_COMPUTE_NO_ENCLAVE_CONFIG,
-                    "Enclave configuration must no be null"
-            );
+                    "Enclave configuration must no be null");
         }
-        if (!enclaveConfig.getValidator().isValid()){
+        if (!enclaveConfig.getValidator().isValid()) {
             throw new TeeSessionGenerationException(
                     APP_COMPUTE_INVALID_ENCLAVE_CONFIG,
                     "Invalid enclave configuration: " +
-                            enclaveConfig.getValidator().validate().toString()
-            );
+                            enclaveConfig.getValidator().validate().toString());
         }
 
-        tokens.put(APP_MRENCLAVE, enclaveConfig.getFingerprint());
+        enclaveBase.mrenclave(enclaveConfig.getFingerprint());
         // extract <IEXEC_INPUT_FILE_NAME_N, name>
         // this map will be empty (not null) if no input file is found
         Map<String, String> inputFileNames = IexecEnvUtils.getComputeStageEnvMap(taskDescription)
                 .entrySet()
                 .stream()
                 .filter(e -> e.getKey().contains(IexecEnvUtils.IEXEC_INPUT_FILE_NAME_PREFIX))
                 .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
-        tokens.put(INPUT_FILE_NAMES, inputFileNames);
+        tokens.putAll(inputFileNames);
```

Comment:
Same as before. We should maybe not store the input file names in a temporary var but use `.forEach` instead.